### ISSUE TITLE
setfacl: Option -m: Invalid argument near character 10

### ISSUE THREAD

**Issue by zefanja** [linuxmuster/linuxmuster-base7#29]
Today I noticed some errors in the `setup.samba-users.log` logfile:

```
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /var/tmp ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /var/lib/samba/printers                                     ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### net conf addshare print$ /var/lib/samba/printers writeable=y guest_ok=n ####
ERROR: share print$ already exists.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /var/lib/samba/printers ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/linbo                                                  ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### net conf addshare linbo /srv/linbo writeable=y guest_ok=n            ####
ERROR: share linbo already exists.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,o::rx" /srv/linbo ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/pgm                       ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### net conf addshare pgm /srv/samba/schools/default-school/pgm writeable=y guest_ok=n ####
ERROR: share pgm already exists.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /srv/samba/schools/default-school/pgm ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/cdrom                     ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### net conf addshare cdrom /srv/samba/schools/default-school/cdrom writeable=y guest_ok=n ####
ERROR: share cdrom already exists.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /srv/samba/schools/default-school/cdrom ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/share                     ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### net conf addshare share /srv/samba/schools/default-school/share writeable=y guest_ok=n ####
ERROR: share share already exists.
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /srv/samba/schools/default-school/share ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/share/classes             ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /srv/samba/schools/default-school/share/classes ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/share/projects            ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:Domain Admins" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rx,o::-" /srv/samba/schools/default-school/share/projects ####
chown: invalid user: ‘global-admin:Domain Admins’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/share/school              ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:users" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:users:rwx,o::-" /srv/samba/schools/default-school/share/school ####
chown: invalid user: ‘global-admin:users’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### mkdir -p /srv/samba/schools/default-school/share/teachers            ####
------------------------------------------------------------------------------
------------------------------------------------------------------------------
#### 2018-07-04 10:38:12                                                  ####
#### /usr/share/linuxmuster/setfacl.sh "global-admin:teachers" "u::rwx,u:global-admin:rwx,g::rwx,g:Domain Admins:rwx,g:teachers:rwx,o::-" /srv/samba/schools/default-school/share/teachers ####
chown: invalid user: ‘global-admin:teachers’
setfacl: Option -m: Invalid argument near character 10
------------------------------------------------------------------------------

```

**Comment by jeffbeck** [comment]
Why does linuxmuster-base set acl's under /srv/samba?
This is done already by sophomorix when the school, groups, ... are created.

setfacl is also the wrong command.
On samba shares use:  smbcacls

**Comment by HappyBasher** [comment]
@zefanja 
it seems samba provisioning fails before. please try a fresh setup.

**Comment by HappyBasher** [comment]
@jeffbeck 
acls have to be set because
- sysvol share is not user readable
- share folders pgm, cdrom, share/classes, share/school, share/projects, share/teachers do not exist and have therefor to be created by setup.

**Comment by jeffbeck** [comment]
@HappyBasher 

All directories for a school and their NTACL can be created with the sophomorix command:

`# sophomorix-school --school <schoolname>`

**Comment by netzint** [comment]
> acls have to be set because
The acls have to be set by Sophomorix (only).

> sysvol share is not user readable
Thats ok. This is a samba system share and must not be edited. The ACLs of this share is dynamically created via GPO-Objects.

> share folders pgm, cdrom, share/classes, share/school, share/projects, share/teachers do not exist and have therefor to be created by setup.

Sophomorix creates this directories and manage the ALCs on it. 

Please undo this changes as soon as possible.

**Comment by HappyBasher** [comment]
Fixed in 7.0.7.
